### DISCUSSION TITLE
Errors in (200) Pole Figure calculation

### DISCUSSION THREAD

**Discussion by creuzige** [mtex-toolbox/mtex#510] [Ask Anything] [answered]
I am trying to import pole figures to check some ODFs and then export the recalculated pole figures.  It looks like there are some issues with the (200) pole figure.  You can see below the recalculated pole figure intensities aren't in the right spot (also shown in the difference plot)

**Input pole figure**
![InputPF](https://user-images.githubusercontent.com/10198749/90053048-f669cf00-dca7-11ea-9fe3-206a6fbe2aec.png)

**Recalculated pole figure**
![RecalcPF](https://user-images.githubusercontent.com/10198749/90053050-f7026580-dca7-11ea-9c64-053196db3f5c.png)

**Difference pole figure**
![DifferencePF](https://user-images.githubusercontent.com/10198749/90053045-f5d13880-dca7-11ea-810b-6412ceb79e80.png)

I seem to recall the (#00) pole figure calculations were difficult, is there a work around or something I've missed? I've included the script and (hopefully all) the relevant data files in the .zip below.

Using mtex 5.3, Matlab 2019a, MacOS 10.15.6

[GitHub-mtex.zip](https://github.com/mtex-toolbox/mtex/files/5064933/GitHub-mtex.zip)

**Comment by kilir** [comment]
Hi,
if you use
```
pdf_rc_a = calcPoleFigure(odf_austenite.normalize,h_austenite_save,r_save)
```
do polefigures look reasonably similar?

Cheers,
Rüdiger

**Comment by creuzige** [comment]
Gave it a try and got this error:

```
Unrecognized method, property, or field 'normalize' for class 'ODF'.

Error in CHESS_data (line 237)
pdf_rc_a = calcPoleFigure(odf_austenite.normalize,h_austenite_save,r_save)
```

The issue is more in the martensite phase, but I tried this for both of them.  I thought normalize would apply more to pole figures than ODFs...

I'm suspicious that there's something up with the ODF calculation; either I've mislabeled something, the data I've input is insufficient to avoid a wildly different solution, there's some initial condition that's tipping the solution this way, or something else.

**Comment by kilir** [comment, marked as answer]
Oh sorry my bad, slipped the line, I meant computing the odf from the normalized pf
```
odf_martensite = calcODF(pf_martensite.normalize, CrySym{2},SpecSym{1}, 'halfwidth',4*degree, 'resolution',4*degree)
```

**Comment by creuzige** [comment]
See below, that did seem to take care of most of the issues. I had thought the ODF calculation did a pole figure normalization as part of the input, did that change recently?  

**Recalculated Pole Figures after normalizing on ODF calc**
![RecalcPF-afternorm](https://user-images.githubusercontent.com/10198749/90059752-aee84080-dcb1-11ea-9b7a-9206bc9d7dae.png)
**Difference Pole Figures after normalizing on ODF calc**
![DifferencePF-afternorm](https://user-images.githubusercontent.com/10198749/90059756-b0196d80-dcb1-11ea-98aa-d5a29c5c290a.png)
**Original Pole Figures**
![Original-PF](https://user-images.githubusercontent.com/10198749/90060402-a93f2a80-dcb2-11ea-8772-4343d2754408.png)

I'm kind of struck by the pattern in the difference plots, the distribution of r vectors (original pole figures) really comes through there. To improve the accuracy do you think I would I be better off trying to reduce the ODF resolution, or increasing the number of iterations?

Thanks!

**Comment by ralfHielscher** [comment]
Hi Adam,

I also wasn't aware that the scaling hat such a big impact on the result. You can observe this a bit during the computation where you get
```
  0 |  0.35  0.34  0.19
  1 |  0.15  0.27  0.14
  2 |  0.10  0.27  0.11
  3 |  0.09  0.28  0.10
  4 |  0.08  0.30  0.10
  5 |  0.08  0.30  0.10
  6 |  0.08  0.31  0.09
```
indicating that the fit of the second pole figure is much worse then the other two. This is because this pole figure has lower intensities and therefore considered less accurate. You can weight a certain pole figure as more important by multiplying its intensities by some factor. So the important think is not the normalization but the ratio of the intensities between the different pole figures. After normalization you get
```
  0 |  0.35  0.34  0.19
  1 |  0.18  0.19  0.15
  2 |  0.14  0.11  0.14
  3 |  0.13  0.10  0.13
  4 |  0.13  0.09  0.14
  5 |  0.13  0.08  0.14
  6 |  0.13  0.08  0.14
```
indicating that the second pole figure is now fitted better.

Ralf.
